lib/CL/devices/level0/level0-driver.cc

```diff
@@ -0,0 +1,2239 @@
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  ze_device_compute_properties_t ComputeProperties{};
+  ComputeProperties.stype = ZE_STRUCTURE_TYPE_DEVICE_COMPUTE_PROPERTIES;
+  ComputeProperties.pNext = nullptr;
+  Res = zeDeviceGetComputeProperties(DeviceHandle, &ComputeProperties);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  ze_device_module_properties_t ModuleProperties{};
+  ModuleProperties.stype = ZE_STRUCTURE_TYPE_DEVICE_MODULE_PROPERTIES;
+  ModuleProperties.pNext = nullptr;
+  Res = zeDeviceGetModuleProperties(DeviceHandle, &ModuleProperties);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  uint32_t QGroupPropCount = 32;
+  ze_command_queue_group_properties_t QGroupProps[32];
+  for (uint32_t i = 0; i < 32; ++i) {
+    QGroupProps[i].pNext = nullptr;
+    QGroupProps[i].stype = ZE_STRUCTURE_TYPE_COMMAND_QUEUE_GROUP_PROPERTIES;
+  }
+  Res = zeDeviceGetCommandQueueGroupProperties(DeviceHandle, &QGroupPropCount,
+                                               QGroupProps);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  uint32_t MemPropCount = 32;
+  ze_device_memory_properties_t MemProps[32];
+  for (uint32_t i = 0; i < 32; ++i) {
+    MemProps[i].pNext = nullptr;
+    MemProps[i].stype = ZE_STRUCTURE_TYPE_DEVICE_MEMORY_PROPERTIES;
+  }
+  Res = zeDeviceGetMemoryProperties(DeviceHandle, &MemPropCount, MemProps);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  ze_device_memory_access_properties_t MemAccessProperties{};
+  MemAccessProperties.stype = ZE_STRUCTURE_TYPE_DEVICE_MEMORY_ACCESS_PROPERTIES;
+  MemAccessProperties.pNext = nullptr;
+  Res = zeDeviceGetMemoryAccessProperties(DeviceHandle, &MemAccessProperties);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  uint32_t CachePropCount = 32;
+  ze_device_cache_properties_t CacheProperties[32];
+  for (uint32_t i = 0; i < 32; ++i) {
+    CacheProperties[i].pNext = nullptr;
+    CacheProperties[i].stype = ZE_STRUCTURE_TYPE_DEVICE_CACHE_PROPERTIES;
+  }
+  Res = zeDeviceGetCacheProperties(DeviceHandle, &CachePropCount,
+                                   CacheProperties);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  ze_device_image_properties_t ImageProperties{};
+  ImageProperties.stype = ZE_STRUCTURE_TYPE_DEVICE_IMAGE_PROPERTIES;
+  ImageProperties.pNext = nullptr;
+  Res = zeDeviceGetImageProperties(DeviceHandle, &ImageProperties);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  ze_device_external_memory_properties_t ExternalMemProperties{};
+  ExternalMemProperties.stype =
+      ZE_STRUCTURE_TYPE_DEVICE_EXTERNAL_MEMORY_PROPERTIES;
+  ExternalMemProperties.pNext = nullptr;
+  Res =
+      zeDeviceGetExternalMemoryProperties(DeviceHandle, &ExternalMemProperties);
+  if (Res != ZE_RESULT_SUCCESS)
+    return;
+
+  POCL_MSG_PRINT_LEVEL0("all device info collected\n");
+
+  /**************************************************************************/
+
+  // TODO
+  ClDev->profiling_timer_resolution = 1;
+
+  // Fixed props
+  {
```

Comment:
Why the additional code segment {}?

Reply:
only to make code navigation a bit easier (it's a ~600 line function) and "group" logically related code. Should i remove it?

Reply:
Yep, looks a bit strange. Just change it to a "comment bar" (which you liked to use elsewhere) instead?

Reply:
This code segment is still there?

Reply:
removed